nemo/collections/speechlm/data/dataset/audio_text_dataset.py

```diff
@@ -0,0 +1,1039 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
```

Comment:
IMO we should only support Lhotse based dataset for this new collection to reduce the amount of complexity and maintenance. Supporting two completely different dataloader implementations will split our resources and waste our energy going forward. I suggest that this whole file and references to it throughout the collection should be removed.

Reply:
I think we should keep the non-lhotse dataset as well. It only needs to have the basic functions so that people not familiar with lhotse can still know what's happning in the dataset. For more advanced features we will just ask users to use lhotse dataloader.